packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -363,70 +316,15 @@ export class NxMonorepoProject extends TypeScriptProject {
       });
     }
 
-    options.nxConfig?.nxCloudReadOnlyAccessToken && this.addDevDeps("nx-cloud");
-
-    new IgnoreFile(this, ".nxignore").exclude(
-      "test-reports",
-      "target",
-      ".env",
-      ".pytest_cache",
-      ...(this.nxConfig?.nxIgnore || [])
-    );
-
-    this.nxJson = new JsonFile(this, "nx.json", {
-      obj: {
-        extends: "nx/presets/npm.json",
-        npmScope: "monorepo",
-        tasksRunnerOptions: {
-          default: {
-            runner: options.nxConfig?.nxCloudReadOnlyAccessToken
-              ? "nx-cloud"
-              : "nx/tasks-runners/default",
-            options: {
-              useDaemonProcess: false,
-              cacheableOperations: options.nxConfig?.cacheableOperations || [
-                "build",
-                "test",
-              ],
-              accessToken: options.nxConfig?.nxCloudReadOnlyAccessToken,
-            },
-          },
-        },
-        namedInputs: {
-          // https://nx.dev/more-concepts/customizing-inputs#defaults
-          default: ["{projectRoot}/**/*"],
-          ...options.nxConfig?.namedInputs,
-        },
-        targetDefaults: {
-          build: {
-            dependsOn: ["^build"],
-          },
-          ...this.nxConfig?.targetDefaults,
-        },
-        affected: {
-          defaultBase: this.nxConfig?.affectedBranch || "mainline",
-        },
-      },
-    });
-  }
-
-  /**
-   * Ensure NXProject is added when adding subprojects to the monorepo.
-   *
-   * @internal
-   * @param subproject project to add.
-   */
-  _addSubProject(subproject: Project) {
-    !NxProject.of(subproject) &&
-      subproject._addComponent(new NxProject(subproject));
-    super._addSubProject(subproject);
+    this.nx = NxWorkspace.of(this) || new NxWorkspace(this);
```

Comment:
Can we also have an ensure method in NxWorkspace?

Reply:
NxWorkspace should only be set once by the root, and is not required. While do to construct/synth order of operations, the NxProject might not have been instantiated when it is needed - so `ensure` makes sense. I don't think we want the same loose functionality around NxWorkspace.

But if have a use case for it then yes is easy to add

Reply:
Makes sense, please ignore.